src/components/button/types.ts

```diff
@@ -0,0 +1,19 @@
+import { ReactElement } from 'react'
+
+type ButtonElementProps = JSX.IntrinsicElements['button']
+
+// TODO: add jsdoc to each prop
+export type ButtonProps = Pick<
```

Comment:
I think this should be an `interface`! Especially since its extending the button element props.

Reply:
Interesting notes in[ the comments](https://www.typescriptlang.org/play#example/types-vs-interfaces) here re: interfaces v. types

Reply:
It's up to you though 👐

Reply:
Just pushed up some commits that have the outcome of making this an `interface`. 😊